src/go/k8s/internal/client/users/client.go

```diff
@@ -0,0 +1,203 @@
+package users
+
+import (
+	"context"
+	"slices"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/src/go/k8s/api/redpanda/v1alpha2"
+)
+
+// Client is a high-level client for managing users in a Redpanda cluster.
+type Client struct {
+	kafkaAdminClient  *kadm.Client
+	adminClient       *rpadmin.AdminAPI
+	client            client.Client
+	generator         *passwordGenerator
```

Comment:
I'm going to opt, at least for now, to keep the Kubernetes client within the user client --  especially since there's already some code testing the Kubernetes part of the password generation. For doing more direct testing of the creation routines, I'm pretty much doing a no-no and calling the unexported `create` implementation directly with a password, so that way I can separate out tests for the API calls v. the secret creation.

Ideally I think you're right and I could refactor out some sort of password store and pass in an interface, but I think this is readable/tested well enough for now that this could be a later pass.